agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -121,6 +148,8 @@ def apply
       container = nil
 
       exclusive {
+        apply_started!
```

Comment:
Could this `apply_started!` / `apply_finished!` be refactored into a block `apply!` that internally handles settings timestamps?

Reply:
I don't immediately see how that could be refactored sensibly... this is already in the method named  `apply`, and it's not used anywhere else.

Reply:
```ruby
def apply!
  apply_started!
  yield
rescue
  raise
else
  apply_finished!
end
```

```ruby
exclusive do
  apply! do
  end
end
```

? :)

Reply:
Yeah, but I think that just ends up being less clear...?

Reply:
`apply!` is probably bad name (or `apply` should be renamed). Anyway with block you don't need to expose `apply_started!` and `apply_finished!` methods at all?

Reply:
That said I'm ok with merging current implementation. I was only pointing out that this current logic adds a bit unnecessary noise to the class.

Reply:
I agree that `apply` wants some refactoring, but I don't think the `apply_started/finished!` would be the place to start... it needs something else to simplify the `exclusive`/`applying` block entirely